src/sparseml/pytorch/torchvision/train.py

```diff
@@ -415,21 +417,51 @@ def collate_fn(batch):
         alpha = 1.0 - args.model_ema_decay
         alpha = min(1.0, alpha * adjust)
         model_ema = utils.ExponentialMovingAverage(
-            model_without_ddp, device=device, decay=1.0 - alpha
+            model, device=device, decay=1.0 - alpha
         )
 
-    if args.resume:
-        checkpoint = torch.load(args.resume, map_location="cpu")
-        model_without_ddp.load_state_dict(checkpoint["model"])
-        if not args.test_only:
-            optimizer.load_state_dict(checkpoint["optimizer"])
-            lr_scheduler.load_state_dict(checkpoint["lr_scheduler"])
+    if args.checkpoint_path:
+        checkpoint = _load_checkpoint(args.checkpoint_path)
+        model.load_state_dict(checkpoint["model"])
+        optimizer.load_state_dict(checkpoint["optimizer"])
+        lr_scheduler.load_state_dict(checkpoint["lr_scheduler"])
+        if model_ema:
+            model_ema.load_state_dict(checkpoint["model_ema"])
+        if scaler:
+            scaler.load_state_dict(checkpoint["scaler"])
+
         args.start_epoch = checkpoint["epoch"] + 1
+
+        manager = ScheduledModifierManager.from_yaml(args.recipe_path)
+        checkpoint_manager = ScheduledModifierManager.from_yaml(
+            checkpoint["checkpoint_recipe"]
+        )
+
+        checkpoint_manager.apply_structure(model, epoch=checkpoint["epoch"])
+    elif args.resume:
+        checkpoint = _load_checkpoint(args.resume)
+        model.load_state_dict(checkpoint["model"])
+        optimizer.load_state_dict(checkpoint["optimizer"])
+        lr_scheduler.load_state_dict(checkpoint["lr_scheduler"])
         if model_ema:
             model_ema.load_state_dict(checkpoint["model_ema"])
         if scaler:
             scaler.load_state_dict(checkpoint["scaler"])
 
+        args.start_epoch = checkpoint["epoch"] + 1
+
+        # NOTE: override manager with the checkpoint's manager
+        manager = ScheduledModifierManager.from_yaml(checkpoint["checkpoint_recipe"])
+        checkpoint_manager = None
+
+        manager.initialize(model, epoch=checkpoint["epoch"])
```

Comment:
to clarify - model state dict load in resume should come after the checkpoint recipe is initialized (ie after graph state is reproduced)

Reply:
I think I did this right - can you verify?

Reply:
looks good - see comment about potentially refactoring